src/context.rs

```diff
@@ -44,4 +52,27 @@ impl Context {
 
         Ok(::device::new(self, ptr))
     }
+
+    /// Creates a device for a given major/minor device number pair
+    pub fn device_from_devnum(&self, kind: ::DeviceType, major: u64, minor: u64) -> ::Result<Device> {
```

Comment:
Why not accept `dev_t` as an argument instead of separate major/minor numbers? If the user already has a `dev_t`, this API forces them to unpack the `dev_t` into a major and minor number to pass to this function, only to internally reconstruct a `dev_t` to pass to `udev_device_new_from_devnum()`. I think this library can deal with `dev_t` as an opaque type.